support/index.html

```diff
@@ -3,45 +3,163 @@
+        'hide',
+        'unrecoverable_error',
+        'authenticated',
+        'authorization_error',
+        'hash_parsed',
+        'signin ready',
+        'signup ready',
+        'forgot_password ready',
+        'forgot_password submit',
+        'signin submit',
+        'signup submit',
+        'federated login'
+      ];
+      validEvents.forEach(function (e) {
+        instance.on(e, function () {
+          var args = arguments;
+          if (arguments.length === 1) {
+            args = arguments[0];
+          }
+          logs.push({ event: e, arguments: args });
+          printLogs();
+        });
+      });
+    }
+    $('#btn-clear-log').on('click', clearLogs);
+    $(function () {
+      printLogs();
+    });
+  </script>
   <script>
-    const cid = "BWDP9XS89CJq1w6Nzq7iFOHsTh6ChS2b";
+    const clientId = "BWDP9XS89CJq1w6Nzq7iFOHsTh6ChS2b";
     const domain = "brucke.auth0.com";
-    const options = {
+    const defaultOptions = {
       allowShowPassword: true,
       usernameStyle: 'email',
       defaultDatabaseConnection: 'acme',
       prefill: {
         email: 'johnfoo@gmail.com'
-      }
+      },
+      passwordlessMethod: 'code',
+      allowedConnections: ['sms', 'acme']
     };
-
-    const lock = new Auth0Lock(cid, domain, options);
-    window.lock = lock;
-
-    lock.on("authenticated", function (authResult) {
-      console.log(authResult);
-
-      if (!authResult.accessToken) return;
-
-      lock.getUserInfo(authResult.accessToken, function (error, profile) {
-        console.log(error, profile);
+    function initLock() {
+      var lock = new Auth0Lock(clientId, domain, defaultOptions);
+      window.localStorage.lastUsed = 'lock'
+      subscribeToEvents(lock);
+      return lock;
+    }
+    function initPasswordless() {
+      var lockPasswordless = new Auth0LockPasswordless(clientId, domain, defaultOptions);
+      window.localStorage.lastUsed = 'passwordless'
+      subscribeToEvents(lockPasswordless);
+      return lockPasswordless;
+    }
+    $(function () {
+      $('#btn-show-lock').on('click', function () {
+        clearLogs();
+        var lock = initLock();
+        lock.show({
+          languageDictionary: {
+            title: 'Lock',
+          }
+        });
+      });
+      $('#btn-show-passwordless').on('click', function () {
+        clearLogs();
+        var lockPasswordless = initPasswordless();
+        lockPasswordless.show({
+          languageDictionary: {
+            title: 'Lock + Passwordless',
+          }
+        });
       });
-    });
 
-    lock.on("authorization_error", function (error) {
-      console.log('authorization_error', error);
+      //make sure we initialize Lock so we can parse the hash
+      var lastUsed = window.localStorage.lastUsed;
+      if (!lastUsed) {
+        return;
+      }
+      if (lastUsed === 'lock') {
+        initLock();
+      } else {
+        initPasswordless();
+      }
+      window.localStorage.removeItem('lastUsed');
```

Comment:
Noob asking. What happens if I refresh the page? Won't it mess the state?

Reply:
yeah, but that's fine. it's only there to instantiate lock/lockpasswordless in the callback page. if you refresh twice then you have to start over. this page is for internal usage only, so that's fine